### PR TITLE
added servisec, interfaces, valodation

### PR DESCRIPTION
**Pull request by Fanar1n**

### CODE REVIEW

File: Bank/Bank.BLL/Infrastructure/Validation.cs [hunk 1]
```diff
@@ -37,14 +39,7 @@ public bool DataValidationOwnerSecondName(CreditCard creditCard)
...
         {
-            var creditCard = _db.CreditCard.AsNoTracking().FirstOrDefault(p => p.Id == id);
-
-            if (creditCard == null)
-            {
-                return true;
-            }
-
-            return false;
+            return _creditCardRepository.GetAll().FirstOrDefault(p => p.Id == id) is not null;
```
Comment (Blackolf931) [inline Bank/Bank.BLL/Infrastructure/Validation.cs:42]: Why we need that FirstOrDefault? maybe check only null or check length

File: Bank/Bank.BLL/Interfaces/ICreditCardServices.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Bank.BLL.Interfaces
+{
+    public interface ICreditCardServices<CreditCard>
```
Comment (stas1408123) [inline Bank/Bank.BLL/Interfaces/ICreditCardServices.cs:3]: For what you need Generic?

File: Bank/Bank.BLL/Services/CreditCardServices.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            if (!_validation.DataValidationId(id))
+            {
+                throw new ArgumentException("Data or Id is not correct");
+            }
+
+            _creditCardRepository.Delete(id);
+        }
+
+        private CreditCardEntity ConvertCreditCardToCreditCardEntity(CreditCard item)
```
Comment (stas1408123) [inline Bank/Bank.BLL/Services/CreditCardServices.cs:83]: Create class for mapper

File: Bank/Bank.DAL/Interfaces/ICreditCardRepository.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Bank.DAL.Interfaces
+{
+    public interface ICreditCardRepository<CreditCardEntity>
```
Comment (stas1408123) [inline Bank/Bank.DAL/Interfaces/ICreditCardRepository.cs:3]: For what you need Generic?

File: Bank/Bank.BLL/Infrastructure/Validation.cs (hunk 1 above)
Comment (stas1408123) [inline Bank/Bank.BLL/Infrastructure/Validation.cs:42]: ??

**Review by stas1408123 — CHANGES_REQUESTED** [review]

File: Bank/Bank.BLL/Infrastructure/Validation.cs (hunk 1 above)
Comment (hromau) [reply Bank/Bank.BLL/Infrastructure/Validation.cs:42]: Any()

**Review by hromau — CHANGES_REQUESTED** [review]

File: Bank/Bank.BLL/AMapper/Mapper.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
+﻿using Bank.BLL.Models;
+using Bank.DAL.Entities;
+
+namespace Bank.BLL.AMapper
```
Comment (Blackolf931) [inline Bank/Bank.BLL/AMapper/Mapper.cs:4]: `AMapper?` maybe we can name folder as Mappers?

**Review by Blackolf931 — APPROVED** [review]

File: Bank/Bank.BLL/Infrastructure/Validation.cs [hunk 6]
```diff
@@ -37,14 +39,14 @@ public bool DataValidationOwnerSecondName(CreditCard creditCard)
 
         public bool DataValidationId(int id)
```
Comment (hromau) [inline Bank/Bank.BLL/Infrastructure/Validation.cs:40]: make naming more understandable. like IsCardExist, or IsCvvValid and etc

**Review by hromau — CHANGES_REQUESTED** [review]

**Review by stas1408123 — APPROVED** [review]

**Review by hromau — APPROVED** [review]

**Pull request merged**